src/NServiceBus.Core/Transports/Learning/LearningTransportMessagePump.cs

```diff
@@ -0,0 +1,245 @@
+                string bodyPath;
+                Dictionary<string, string> headers;
+                ExtractMessage(out bodyPath, message, out headers);
+
+                string ttbrString;
+
+                if (headers.TryGetValue(Headers.TimeToBeReceived, out ttbrString))
+                {
+                    var ttbr = TimeSpan.Parse(ttbrString);
+                    //file.move preserves create time
+                    var sentTime = File.GetCreationTimeUtc(transaction.FileToProcess);
+
+                    if (sentTime + ttbr < DateTime.UtcNow)
+                    {
+                        await transaction.Commit()
+                            .ConfigureAwait(false);
+                        return;
+                    }
+                }
+                var tokenSource = new CancellationTokenSource();
+
+                var context = new ContextBag();
+                context.Set(transaction);
+
+                var body = await AsyncFile.ReadBytes(bodyPath, cancellationToken)
+                    .ConfigureAwait(false);
+
+                var transportTransaction = new TransportTransaction();
+
+                transportTransaction.Set(transaction);
+
+                var messageContext = new MessageContext(messageId, headers, body, transportTransaction, tokenSource, new ContextBag());
+
+                try
+                {
+                    await onMessage(messageContext)
+                        .ConfigureAwait(false);
+                }
+                catch (Exception exception)
+                {
+                    transaction.ClearPendingOutgoingOperations();
+                    var processingFailures = retryCounts.AddOrUpdate(messageId, id => 1, (id, currentCount) => currentCount + 1);
+
+                    var errorContext = new ErrorContext(exception, headers, messageId, body, transportTransaction, processingFailures);
+
+                    var actionToTake = await onError(errorContext)
+                        .ConfigureAwait(false);
+
+                    if (actionToTake == ErrorHandleResult.RetryRequired)
+                    {
+                        transaction.Rollback();
+                        return;
+                    }
+                }
+
+                if (tokenSource.IsCancellationRequested)
+                {
+                    transaction.Rollback();
+                    return;
+                }
+
+                await transaction.Commit()
+                    .ConfigureAwait(false);
+            }
+            catch (Exception)
+            {
+                transaction.Rollback();
+            }
+        }
+
+        static void ExtractMessage(out string bodyPath, string message, out Dictionary<string, string> headers)
+        {
+            var splitIndex = message.IndexOf(Environment.NewLine, StringComparison.Ordinal);
+            bodyPath = message.Substring(0, splitIndex);
+            var headerStartIndex = splitIndex + Environment.NewLine.Length;
```

Comment:
so you are concerned about someone using two different OS to code against the same solution stored on a shared dir?